packages/vite/src/node/config.ts

```diff
@@ -270,6 +272,7 @@ export async function resolveConfig(
 
   // user config may provide an alternative mode
   mode = config.mode || mode
+  configEnv.mode = mode
```

Comment:
This looks ok here. But I'm a bit confused by the logic of the previous line. 
According to https://github.com/iheyunfei/vite/blob/a7a9a361cf13670c2e241620ddaf15021eda26b5/packages/vite/src/node/config.ts#L103, the inline config `mode` should be the one winning over the config file `mode`, and that is not the case here AFAICS. @AlexandreBonaventure I think you were using this feature, does it works correctly for you when overriding `--mode` in the CLI? It should win over the one you used in your config file.

@iheyunfei as a note, it would be good to have an issue for this kind of PR with a reproduction for reference. It is easier to merge PRs when we have a clear bug that they are fixing.

Reply:
@patak-js  Yes, I'm going to raise an issue. For your question, I look down the codes and It won't win over the mode in the config file.

Reply:
Ya, looks like there are other issues to fix here

Reply:
Fix it too. After looking back, I found I just solve inconstancy in some cases. Still, there is a case that could not be easily fixed. I will open another issue to describe it.